geode-core/src/main/java/org/apache/geode/internal/cache/BucketRegion.java

```diff
@@ -604,17 +611,23 @@ public void handleWANEvent(EntryEventImpl event) {
 
     if (!(this instanceof AbstractBucketRegionQueue)) {
       if (getBucketAdvisor().isPrimary()) {
+        if (notPrimary && needToWaitForColocatedBucketsBecomePrimary()) {
```

Comment:
`nonPrimary` appears to not be synchronized before access here.

Reply:
It is later accessed in needToWaitForColocatedBucketsBecomePrimary() with synchronized on notPrimary.

Reply:
Access should be symmetrical, when it isn't it raises an eyebrow, so even making it volatile still begs why synchronized in some cases and not others.